src/main/java/org/jboss/tools/intellij/analytics/SaUtils.java

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.analytics;
+
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.json.JSONObject;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.File;
+import java.io.Writer;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.util.zip.GZIPInputStream;
+
+public class SaUtils {
+
+    /**
+     * <p>Get Stack Analysis Report for given manifest file using CLI.</p>
+     *
+     * @param filePath Path to target manifest file
+     *
+     * @return A JSONObject having SA Report.
+     *
+     * @throws IOException In case of process failure
+     * @throws InterruptedException In case of process failure
+     */
+    public JSONObject getReport(String filePath) throws IOException, InterruptedException {
+        // Get the SA report using CLI and return JSON data
+        return new JSONObject(new SaProcessExecutor().performStackAnalysis(filePath));
+    }
+
+
+    /**
+     * <p>Open a custom editor window.</p>
+     *
+     * <p>The custom editor window will open a file which will have browser attached to it.</p>
+     *
+     * @param instance An instance of FileEditorManager.
+     * @param url URL to be opened in attached browser.
+     *
+     * @throws IOException In case of process failure
+     */
+    public void openCustomEditor(FileEditorManager instance, String url) throws IOException {
+
+        // Close custom editor if already opened in previous run
+        closeCustomEditor(instance);
+
+        // Create a temp file in sandbox directory which is registered with SaReportEditorProvider.
+        File reportFile = new File(SaConfig.REPORT_FILE_PATH);
```

Comment:
Does not seem a temp file to me. A file with a fixed file name is created in the current directory. Thus if we have projects with both pom.xml and package.json for example, this will fail. So you should create a real temp file with a specific file extension and bind the browser based editor on it.

Reply:
Yes, the name of file is fixed and it is overwritten each time, the reason behind it was that temp file will be created having a random name with given prefix/suffix, only issue with it is that when file is opened in Editor, the name of file would look odd. Is it Okay with you if I create file with fixed name, but inside a unique directory for each execution? so there wont be any conflicts between execution from same project or different project. for Ex $TEMP/ProcesID or UUID/Dependency Analytics Report

PFA screenshot of Editor view in case of temp file name.

![image](https://user-images.githubusercontent.com/63054462/131973183-3e1b081f-edc0-43b9-8c65-dc82ae4d7465.png)

Reply:
Best way is to use a temp file and to change the editor getName() method to report correct information: ex Report for xxx (pom.xml,...)

Reply:
I just tried to rename it using `FileEditor.getName()`, but what I found is that, it can not rename Editor, it changes the name of File tab shown in the bottom if `PLACE_AFTER_DEFAULT_EDITOR` or `PLACE_BEFORE_DEFAULT_EDITOR` `FileEditorPolicy` is used, here I am using `HIDE_DEFAULT_EDITOR` policy so file will have only one tab showing report and getName() will not be useful anymore. Please correct me if I understood it wrong.

![image](https://user-images.githubusercontent.com/63054462/131985533-a023f454-a5d4-4539-89da-abf9ae2a1ef5.png)

Reply:
As you suggested, I have changed it from a fixed file to a temp file which will be created inside temporary directory of system having a random unique name on each execution.

Reply:
Able to rename the file tab using `EditorTabTitleProvider`, it will set title of temp file opened in Editor window without renaming the physical file, hence on rerun from same project/other project new temp file will be created and on opening only title will be changed  in Editor.